### ISSUE TITLE
error reading the configuration file

### ISSUE THREAD

**Issue by raednet** [lashkari/gadget-devtools#2]
Hi Lashari,
We try to install the gadget and get this error, however we add the certificate to trust store but still issue see the below log.
There was an error reading the configuration file. Sun.Security.Validator.ValidatorException: PKIX path building failed: sun.Security.Provider.Certpath.SunCertPathBuilderException: unable to find valid certification path to requested target

devtools, Mar 21, 2016, trustedCertEntry,
Certificate fingerprint (SHA1): 48:07:15:8B:CA:14:C3:54:0B:85:09:C7:7B:98:37:DB:77:39:97:65

**Comment by lashkari** [comment]
You are receiving this error because your OU Campus ESL server does not recognize the HTTPS certificate I'm using (Let's Encrypt).

To resolve this, you or your server administrator will need to add the [Let's Encrypt intermediate certificate(s)](https://letsencrypt.org/certificates/) to your Java keystore (you'll need to install at least Let’s Encrypt Authority X3, but it's probably best to install the X4 backup certificate as well).

To add the certificates, just use the Java keytool app:

```
keytool -import -trustcacerts -keystore /path/to/oucampus/truststore -alias letsencryptX3 -file /path/to/lets-encrypt-x3-cross-signed.pem

keytool -import -trustcacerts -keystore /path/to/oucampus/truststore -alias letsencryptX4 -file /path/to/lets-encrypt-x4-cross-signed.pem
```

**Edit:** Updated to reflect Let's Encrypt's transition to X3 and X4.

**Comment by raednet** [comment]
Dear lashkari,

We have added the other certificates, also added the root X3 certificate but still not working, please see below the list of added certs. We have added before certs for FaceBook, YouTube and other services and all are working fine. What could be missing?

letsencryptauthoristyx2, Mar 22, 2016, trustedCertEntry,
Certificate fingerprint (SHA1): 3A:7A:7D:70:C0:89:97:BC:33:16:17:44:A5:B7:C2:4A:D8:F6:7B:1B
letsencryptauthoristyx1, Mar 22, 2016, trustedCertEntry,
Certificate fingerprint (SHA1): 3E:AE:91:93:7E:C8:5D:74:48:3F:F4:B7:7B:07:B4:3E:2A:F3:6B:F4
twitter, Nov 15, 2015, trustedCertEntry,
Certificate fingerprint (SHA1): 8F:A4:4D:CD:24:4B:3D:D5:E9:D9:80:D4:7F:1F:47:90:8D:32:93:94
youtube, Nov 15, 2015, trustedCertEntry,
Certificate fingerprint (SHA1): EA:03:41:9E:58:E6:32:F3:E3:69:40:B3:A0:57:81:E7:E5:CA:F8:D0
fb, Nov 15, 2015, trustedCertEntry,
Certificate fingerprint (SHA1): 8C:CB:D9:A1:F3:3C:78:C2:2E:F6:EB:1C:CD:4B:F3:39:1B:9A:EE:4E
dstrootcax3, Mar 22, 2016, trustedCertEntry,
Certificate fingerprint (SHA1): DA:C9:02:4F:54:D8:F6:DF:94:93:5F:B1:73:26:38:CA:6A:D7:7C:13
google, Nov 15, 2015, trustedCertEntry,
Certificate fingerprint (SHA1): 70:13:07:E1:54:E2:8F:CC:6C:1C:3B:A2:54:32:C1:F3:71:B8:87:67
devtools, Mar 22, 2016, trustedCertEntry,
Certificate fingerprint (SHA1): 48:07:15:8B:CA:14:C3:54:0B:85:09:C7:7B:98:37:DB:77:39:97:65
ad, May 19, 2015, trustedCertEntry,
Certificate fingerprint (SHA1): E0:74:56:84:A6:CB:9A:09:70:22:D8:57:03:E1:25:97:22:31:37:64

**Comment by lashkari** [comment]
That is very odd.

You've restarted tomcat after adding the certificates, right? And you are certain that you are using the same truststore that OU Campus is loading?

**Comment by raednet** [comment]
Yes we did a restart for tomcat and using same truststore that ou loading. any ways to install that gadget?

**Comment by lashkari** [comment]
I do not have any immediate plans to move away from Let's Encrypt, but you can follow the instructions in the README to run the gadget yourself (it requires NodeJS): https://github.com/lashkari/gadget-devtools#installing-and-running-locally

If you're running it in production, just call `npm start` instead of `npm run dev`
